### ISSUE TITLE
Accessibility - Backoffice - Account menu label in name

### ISSUE THREAD

**Issue by rickdoesburg** [umbraco/Umbraco-CMS#21942]
### Which Umbraco version are you using?

17.2

### Bug summary

While doing a simple WCAG Audit of the backoffice I noticed some bugs.

The button to trigger the account menu shows your initials. These initials are not available in the aria-label on the button that's overwriting the visible content. Therefor the buttons is not accessible for users using speach to trigger actions.

<img width="170" height="101" alt="Image" src="https://github.com/user-attachments/assets/23d279a7-b271-4d21-a40f-f501797389ca" />

### Steps to reproduce

Sign in to the backoffice

### Expected result / actual result

The aria-label contains the visible text to make sure it passes WCAG 2.5.3 (Label in name)

**Comment by dataCenter430** [comment]
@AndyButland  I'd love to handle this issue too

**Comment by AndyButland** [comment]
For this one I'm wondering if actually "Profile options" - as looks to be currently the case from the screenshot - is a better aria label.  It seems to describe the action of the button better than the initials would do.  What do you think @rickdoesburg?

**Comment by rickdoesburg** [comment]
@AndyButland There's nothing wrong with the current label 'Profile options' but it doesn't contain the visible text. The aria-label overwrites the visible label "RD" in this case. If a user using speech software wanted to click this button they'll get an error saying there is no button with the label "RD" as the aria-label "Profile options" has overwitten it in the accessibility tree. Long story short; all that is needed is that the Initials are added at the end of the `aria-label`. I'll see if I can create the PR myself in a second.  

Edit: This seems to be mix from the CMS and UI repo. Not my field of expertise haha. 

I've done some more research leading me to this sentence: "For the purposes of this SC, text should not be considered a visible label if it is used in a symbolic manner, rather than directly "expressing something in human language". Which in turn directs me to the definitions of "text" according to WCAG. "sequence of characters that can be [programmatically determined](https://www.w3.org/WAI/WCAG21/Understanding/label-in-name.html#dfn-programmatically-determined), where the sequence is expressing something in [human language](https://www.w3.org/WAI/WCAG21/Understanding/label-in-name.html#dfn-human-language)".

So on one hand I'd say the initials aren't human language and therefor this SC does not fail and the current label is fine. When you have a profile picture there isn't a label either. On the other hand adding the initials to the label when no image is selected doesn't hurt anyone.

**Comment by nielslyngsoe** [comment]
@rickdoesburg, okay, so what I read is that you have concluded that this is a non-issue?

For me to understand the aspect correctly, your concern was regarding users who verbally navigate the backoffice, who would say something like "Click RD" or "open RD"?

I think that could in general, become a tricky thing to resolve. There are a few things that does not have a written out name on screen, so how do you target a thing if you dont know what it is called?  Not to turn it down, but trying to be realistic about the challenge, I do find that unrealistic for us to cover.

But we can surely improve the label, and any suggestions are welcome — but listing the initials there I'm not sure gives much value, my aspect is regarding the user who can't see, does not get much value our of the text "Open user options for RD", then rather insert the whole user name. but then again, hopefully you know your self. So maybe its more a question of the formulation. should it say "Open current user options"?

These where my thoughts, please bring your aspect :-) and thanks in advance

**Comment by rickdoesburg** [comment]
@nielslyngsoe What you're saying is correct. There's a difference between screen reader users (usually visually impaired) and speech users (physically impaired). Users using speech to navigate the web do Indeed say "Click RD" to trigger a click event on that button. They navigate the web by sight but are unable to use a pointing device for example. Screen reader users use software that read out the DOM/Accessibility tree of the site. So their software will read the aria-label "Profile options" which is perfectly clear but doesn't match the visual label. In their case that doesn't matter as they are navigating based on the screen reader output.

This WCAG criteria doesn't apply to items without a visible label but does to all other visible labels as long as they are 'human language'. Yes, that means there are buttons without visual labels which can't be clicked using speech. They often have to turn on an overlay like “show numbers”, which puts numbers on each actionable control so they can say “Click 7” instead.

Adding the initials to the "Profile options" aria-label doesn't hurt anyone.

**Comment by nielslyngsoe** [comment]
Hi @rickdoesburg, thanks for explaining that.

With that insight, I would like to understand if changing the label to this could work?
"Open user profile for Rick Doesburg (RD)"

Or would it need to be more directly reassbemled with what was written on screen? I guess my question is if its okay that the initials are present in the text, but as in this suggestion appended in parentheses in the end?

Thanks

**Comment by rickdoesburg** [comment]
@nielslyngsoe As long as the visible text "RD" is present in the `aria-label` this is fine!

**Comment by nielslyngsoe** [comment]
@rickdoesburg, good to know, but then I think the suggestion would be a good improvement. feel free to implement that :-)

**Comment by andreaslborg** [comment]
Hi @nielslyngsoe and @rickdoesburg,

I've had a look at this issue and just want to confirm the intended approach before implementing it.

Should the button label be:
1. `aria-label="RD"`
2. `aria-label="Open user profile for Rick Doesburg (RD)"`

From the discussion it sounds like option 2 is preferred, but I wanted to confirm.

Also regarding the current localization term `visuallyHiddenTexts_profileOptions` that is used for the label, should this be updated to include the user name/initials, or replaced with a new localization key?

**Comment by rickdoesburg** [comment]
Option 2 is the preferred fix except for the word ‘open’. That shouldn’t be included.

**Comment by andreaslborg** [comment]
@rickdoesburg PR is ready #22117, would you mind verifying?
<img width="316" height="243" alt="Image" src="https://github.com/user-attachments/assets/39267eb1-44c8-462a-89cd-3fd5d19d43f9" />

**Comment by rickdoesburg** [comment]
@andreaslborg Yep, looks good. Reason we shouldn't add the word 'open' is because if the menu is open the label doesn't accurately represent the action, which is closing the menu in that case.

**Comment by AndyButland** [comment]
Fixed via https://github.com/umbraco/Umbraco-CMS/pull/22117 for 17.4.
